solr/solrj/src/java/org/apache/solr/common/cloud/ZkStateReader.java

```diff
@@ -635,6 +669,8 @@ public Object getUpdateLock() {
 
   public void close() {
     this.closed  = true;
+    notifications.shutdownNow();  // interrupt
```

Comment:
@markrmiller and I went on a hunt a while ago to try to remove most thread interruptions from Solr due to certain Lucene NIO operations getting permanently wedged due to interrupts.  Is this necessary?